client/src/pages/Booking/Booking.tsx

```diff
@@ -0,0 +1,196 @@
+  const [pastBookings, setPastBookings] = useState<Request[]>([]);
+  const [bookingDays, setBookingDays] = useState<string[]>([]);
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [loading, setLoading] = useState<undefined | boolean>(undefined);
+  const [date, setDate] = useState(new Date() as MaterialUiPickersDate);
+
+  const saveRequests = (requests: Request[]) => {
+    setRequests(requests);
+  };
+
+  useEffect(() => {
+    let ignore = true;
+
+    const initBookingDays = (requests: Request[]) => {
+      const dates = requests.map((request) => {
+        return moment(request.start).format('MM/DD/YYYY');
+      });
+      setBookingDays(dates);
+      const nxtBkn: Request[] = [];
+      const crntBkns: Request[] = [];
+      const pstBkns: Request[] = [];
+      const selectedDate = moment(date).startOf('day');
+
+      requests.forEach((booking) => {
+        const startDate = moment(booking.start).startOf('day');
+        if (startDate.diff(selectedDate, 'days') > 0) {
+          nxtBkn.push(booking);
+        } else if (startDate.diff(selectedDate, 'days') === 0) {
+          crntBkns.push(booking);
+        } else {
+          pstBkns.push(booking);
+        }
+      });
+      setNextBooking(nxtBkn);
+      setCurrentBookings(crntBkns);
+      setPastBookings(pstBkns);
+    };
+
+    async function getRequests() {
+      // send request to back end
+      setLoading(true);
+      const response = await listRequests();
+
+      if (ignore && response && response.requests) {
+        saveRequests(response.requests);
+        initBookingDays(response.requests);
+      }
+      setLoading(false);
+    }
+    getRequests();
+    return () => {
+      ignore = false;
+    };
+  }, [date]);
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  const handleClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const onDateChange = (date: MaterialUiPickersDate): MaterialUiPickersDate => {
+    setDate(date);
+    return date;
+  };
+
+  return (
+    <>
+      <NavBar />
+
+      <Grid container className={classes.root}>
+        <Grid item md={6} className={classes.leftContainer}>
+          <Box className={classes.bookingsBox}>
+            <Card component={Paper}>
+              <CardHeader title="next bookings:" />
+
+              <CardContent className={classes.bookingList}>
+                {loading === false &&
+                  nextBooking.map((request, index) => (
+                    <BookingItem
+                      _id={request._id}
+                      fullName={request.sitterId.username}
+                      start={request.start}
+                      end={request.end}
+                      status={request.status}
+                      username={request.ownerId}
+                      sitterId={request.sitterId._id}
+                      key={index}
```

Comment:
Using indexes as keys is an anti-pattern in React, consider using the requestId instead :)

Ref: https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9